core/vm/contracts.go

```diff
@@ -691,7 +729,7 @@ func (c *blockNumberFromHeader) Run(input []byte, caller common.Address, evm *EV
 	var header types.Header
 	err = rlp.DecodeBytes(input, &header)
 	if err != nil {
-		return nil, gas, err
+		return nil, gas, ErrInputDecode
```

Comment:
I'm worried that returning a different error will mean we won't log the more informative error

Reply:
Interesting, why do they need to be deterministic?

Reply:
I had just assumed error messages to be included under the umbrella of EVM determinism, but maybe they aren't. If not, I'm happy to put the rlp errors back.